components/icu4x_ecma402/src/pluralrules.rs

```diff
@@ -0,0 +1,340 @@
+    use ecma402_traits::pluralrules::Options;
+    use icu_plurals::{PluralCategory, PluralOperands, PluralRuleType};
+    use std::cmp::{max, min};
+
+    // The difference between `first` and `second`, clamped on the bottom by zero.
+    fn clamp_diff(first: usize, second: usize) -> usize {
+        match first > second {
+            true => first - second,
+            false => 0,
+        }
+    }
+
+    /// Converts the trait style option to an equivalent ICU4X type.
+    pub fn to_icu4x_type(style: &Type) -> PluralRuleType {
+        match style {
+            Type::Ordinal => PluralRuleType::Ordinal,
+            Type::Cardinal => PluralRuleType::Cardinal,
+        }
+    }
+
+    /// Formats `n` as a string representation of fixed decimal, based on the supplied options.
+    pub fn fixed_format(n: f64, opts: &Options) -> String {
+        // n = 1234.5
+
+        // 1234.5 -> "1234"
+        // -1234.5 -> "1234"
+        let int_part = format!("{}", n.abs().floor());
+
+        // 5 -> ""
+        // 0.4 -> "4"
+        // 0.43 -> "43"
+        let raw_frac_part = format!("{}", n.fract()); // 0.5
+        let frac_part = if n.fract() == 0.0 {
+            ""
+        } else {
+            &raw_frac_part[2..]
+        };
+
+        dbg!("--> frac={}, fracf={}", &frac_part, &raw_frac_part);
+        dbg!("int_part='{}'; frac_part='{}'", &int_part, &frac_part);
+        dbg!("opts={:?}", opts);
+
+        // Limit the min and max display digits first by individual field.
+        let display_integer_digits = max(int_part.len(), opts.minimum_integer_digits as usize);
+        let display_fraction_digits = max(
+            min(frac_part.len(), opts.maximum_fraction_digits as usize),
+            opts.minimum_fraction_digits as usize,
+        );
+        let total_significant_digits = max(
+            opts.minimum_significant_digits as usize,
+            min(
+                opts.maximum_significant_digits as usize,
+                frac_part.len() + int_part.len(),
+            ),
+        );
+
+        let significant_digits_in_fraction = clamp_diff(total_significant_digits, int_part.len());
+        dbg!(
+            "did={}; dfd={}; sd={}; rsd={}",
+            display_integer_digits,
+            display_fraction_digits,
+            total_significant_digits,
+            significant_digits_in_fraction
+        );
+
+        // Integer fragment.
+        let leading_zeros = clamp_diff(display_integer_digits, int_part.len());
+        let trailing_zeros_in_int_part = clamp_diff(int_part.len(), total_significant_digits);
+        let i = std::iter::repeat('0')
+            .take(leading_zeros)
+            .chain(int_part.chars().take(total_significant_digits))
+            .chain(std::iter::repeat('0').take(trailing_zeros_in_int_part));
+
+        // Decimal dot is printed only if decimals will follow.
+        let dd = std::iter::repeat('.').take(if display_fraction_digits == 0 { 0 } else { 1 });
```

Comment:
that seems like a pretty elaborate way to achieve that. I suggest simplifying it for readers.

Reply:
Well this whole code segment is a bit bizarre, as I tried to avoid rewriting a mutable byte array.  `format!` almost, but not quite fits the levers that `FixedDecimal` offers, so I couldn't use it.  Maybe I'm missing a better way.
Made this particular segment somewhat less surprising.

Reply:
What I meant is that you don't really repeat. It's either one dot or none.
So you could do some sort of `let dd = if ... { &['.'] } else { &[] }` kind of logic, but your solution also looks good.